src/macrofier/engine.mjs

```diff
@@ -1133,45 +1191,49 @@ function insertExampleMacros(template, examples, method, json, templates) {
   return template.replace(/\$\{method\.examples\}/g, content)
 }
 
-function generateResult(result, json, templates) {
-  const type = types.getSchemaType(result, json, { destination: state.destination, section: state.section  })
+function generateResult(schema, json, templates) {
+
+  let name = schema.name || schema.title
+  let result = schema.schema ? schema.schema : schema
```

Comment:
We shouldn't add this code to any methods. I know I left some inspiration to do this in other places, but it makes the code harder to maintain if the `result` parameter could either be a JSON-Schema schema or an OpenRPC Content Descriptor object (https://spec.open-rpc.org/#content-descriptor-object).

We should keep the `result` parameter as a JSON-Schema schema, and pass the `name` in as an optional parameter, ideally via an `options` object.

Reply:
removed